### ISSUE TITLE
etesync.h redefines existing type, causing compiler error

### ISSUE THREAD

**Issue by danvratil** [etesync/etesync-rs#6]
The etesync.h header file contains invalid code:

```
enum EteSyncErrorCode {
  ...
  ...
};
typedef uint32_t EteSyncErrorCode
```

This is not a valid code, since the typedef is re-defining an existing EteSyncErrorCode type, leading to a compiler error:

```
/data/install/kde/include/etesync/../etesync/etesync.h:43:18: error: typedef redefinition with different types ('uint32_t' (aka 'unsigned int') vs 'EteSyncErrorCode')
typedef uint32_t EteSyncErrorCode;
^
/data/install/kde/include/etesync/../etesync/etesync.h:27:6: note: previous definition is here
enum EteSyncErrorCode {
```

**Comment by tasn** [comment]
Maybe that's a C++ thing, but it's valid in C, because in C `enum Foo` is not the same as `Foo` for all I remember. I'll take a look into fixing that in a moment. It's a mistake regardless, I wonder why it generates it this way.

**Comment by tasn** [comment]
I believe I fixed it in bc30859de5157d377c1623cf03d36661f9e7bda7, it was indeed a C++ thing.
